google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java

```diff
@@ -260,6 +263,48 @@ public void downloadTo(Path path) {
     downloadTo(path, new BlobSourceOption[0]);
   }
 
+  /**
+   * Uploads the given file path to this blob using specified blob write options.
+   *
+   * @param path file to be uploaded
+   * @param options blob write options
+   * @throws StorageException upon failure
+   */
+  public void uploadFrom(Path path, BlobWriteOption... options) {
+    uploadFrom(path, DEFAULT_CHUNK_SIZE, options);
+  }
+
+  /**
+   * Uploads the given file path to this blob using specified blob write options
+   * and the provided chunk size.
+   *
+   * @param path file to be uploaded
+   * @param chunkSize the minimum size that will be written by a single RPC.
+   * @param options blob write options
+   * @throws StorageException upon failure
+   */
+  public void uploadFrom(Path path, int chunkSize, BlobWriteOption... options) {
+    if (!Files.exists(path)) {
+      throw new StorageException(0, path + ": No such file");
+    }
+    if (Files.isDirectory(path)) {
+      throw new StorageException(0, path + ": Is a directory");
+    }
+    chunkSize = Math.max(chunkSize, 262144); // adjust with MinChunkSize of BaseWriteChannel
```

Comment:
I am not sure how exactly this works, but what if user wants to use `chuckSize` of less than `262144`. How can he achieve that? Or is that an invalid scenario?

Reply:
262144 is a minimum size of BaseWriteChannel
if we pass 1024 to BaseWriteChannel it will be turned to 262144

Reply:
This is in bytes, right? 
Don't you think we are ignoring the user input if this is less than 262144... If this is how design is then it must be ok. 

Personally I think we should add a JavaDoc note similar to this:

```java
/**
 * ...
 * The minimum `chunkSize` this client supports is 256KB.
 */
```

Reply:
I added a paragraph describing chunk sizes